tests/docker_for_galaxy.py

```diff
@@ -0,0 +1,36 @@
+from collections import namedtuple
+
+import docker
+import pytest
+
+from ephemeris.sleep import wait
+
+GALAXY_IMAGE = "bgruening/galaxy-stable:dev"
+
+client = docker.from_env()
+
+GalaxyContainer = namedtuple('GalaxyContainer', ['url', 'container', 'attributes'])
+
+
+@pytest.fixture
+def start_container(**kwargs):
+    # We start a container from the galaxy image. We detach it. Port 80 is exposed to the host at a random port.
+    # The random port is because we need mac compatibility. On GNU/linux a better option would be not to expose it
+    # and use the internal ip address instead.
+    # But alas, the trappings of a proprietary BSD kernel compel us to do ugly workarounds.
+
+    container = client.containers.run(GALAXY_IMAGE, detach=True, ports={'80/tcp': None}, **kwargs)
+    container_id = container.attrs.get('Id')
+    print(container_id)
+
+    # This seems weird as we also can just get container.attrs but for some reason
+    # the network settings are not loaded in container.attrs. With the get request
+    # these attributes are loaded
+    container_attributes = client.containers.get(container_id).attrs
+
+    # Venturing into deep nested dictionaries.
+    exposed_port = container_attributes.get('NetworkSettings').get('Ports').get('80/tcp')[0].get('HostPort')
+
+    container_url = "http://localhost:{0}".format(exposed_port)
+    wait(container_url, timeout=60)  # We are only going to wait 60 seconds. These are tests, and we are impatient!
```

Comment:
Alternative here is to use planemo, that'd be much speedier and we could use sudo: false on travis.

Reply:
That is a great suggestion. 
However, I am going to wait for someone else to implement this. ;-)
An advantage of the docker containers is that they are easily removed by using the API in docker py. So they do not clutter your system while running locally. Maybe this can also be done with planemo. But I do not use planemo a lot, so someone else is more suited to implement this.
Shall we revisit after this pull request?

Reply:
Sure